build-system/tasks/check-links.js

```diff
@@ -166,13 +166,18 @@ function filterWhitelistedLinks(markdown) {
  * @param {string} markdownFile Path of markdown file, relative to src root.
  * @return {Promise} Used to wait until the async link checker is done.
  */
-function runLinkChecker(markdownFile) {
+async function runLinkChecker(markdownFile) {
+  // The extension generator markdown template has placeholder links that can't
+  // be checked.
+  if (/doc-template\.md$/.test(markdownFile)) {
+    return;
+  }
   // Skip files that were deleted by the PR.
   if (!fs.existsSync(markdownFile)) {
-    return Promise.resolve();
+    return;
   }
-  const markdown = fs.readFileSync(markdownFile).toString();
-  const filteredMarkdown = filterWhitelistedLinks(markdown);
+  const markdown = (await fs.readFile(markdownFile)).toString();
```

Comment:
Why make this an await async vs a readSync? Seems less readable (requires extra parentheses), and it forces this function to become async, which it doesn't need to be

Reply:
Mostly because I wanted to get rid of that pesky `Promise.resolve`. But I have no strong feelings either way, reverted.